website/server/libs/highlightMentions.js

```diff
@@ -0,0 +1,22 @@
+import {model as User} from '../models/user';
+
+const mentionRegex = new RegExp('\\B@[-\\w]+', 'g');
+
+export async function highlightMentions (text) {
+  const mentions = text.match(mentionRegex);
```

Comment:
can we move this call https://github.com/HabitRPG/habitica/commit/d934d9d759e75cc786163646f44c00e52e4cfc98#diff-ade116c537945ae89e2f5f059d4d577fR199 to a single place so we don't count mentions too as regexps can be expensive?

Reply:
I wasn't sure how/where to handle that best. Do you have a suggestion for where to make this call?

Reply:
hmmm... maybe `highlightMentions()` could return both the text and the mentions? so array of [text, mentions] and use the mentions returned in api.postChat?